internal/vendors/equinix/equinix.go

```diff
@@ -0,0 +1,204 @@
+package equinix
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v53/github"
+	"github.com/metal-toolbox/firmware-syncer/internal/config"
+	"github.com/metal-toolbox/firmware-syncer/internal/inventory"
+	"github.com/metal-toolbox/firmware-syncer/internal/vendors"
+	"golang.org/x/oauth2"
+
+	"github.com/pkg/errors"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/operations"
+	"github.com/sirupsen/logrus"
+	serverservice "go.hollow.sh/serverservice/pkg/api/v1"
+)
+
+// Equinix implements the Vendor interface methods to retrieve Equinix OpenBMC firmware files
+type Equinix struct {
+	firmwares []*serverservice.ComponentFirmwareVersion
+	logger    *logrus.Logger
+	metrics   *vendors.Metrics
+	inventory *inventory.ServerService
+	ghClient  *github.Client
+	dstCfg    *config.S3Bucket
+	dstFs     fs.Fs
+	tmpFs     fs.Fs
+}
+
+func New(ctx context.Context, firmwares []*serverservice.ComponentFirmwareVersion, cfgSyncer *config.Syncer, logger *logrus.Logger) (vendors.Vendor, error) {
+	// RepositoryURL required
+	if cfgSyncer.RepositoryURL == "" {
+		return nil, errors.Wrap(config.ErrProviderAttributes, "RepositoryURL not defined")
+	}
+
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: os.Getenv("GITHUB_TOKEN")},
```

Comment:
It would be worth to use viper to load config and env vars, so it can be consistent with other tools we're building - https://github.com/metal-toolbox/conditionorc/blob/ffb5975cbe9025b797165bf0dd16714f569e8de9/internal/app/config.go#L89 

Another thought would be, what if theres different Github repository tokens to be defined.

Not a blocker for this PR though and can be done separately/when the need arises.

Reply:
This is a good idea. I'll file a ticket to re-work all the env vars across the code base in a consistent way that is done by the other tools.

For now I'll update the `GITHUB_TOKEN` to `GITHUB_OPENBMC_TOKEN`

Reply:
Update the env var to use `GITHUB_OPENBMC_TOKEN` in [c4ac33f](https://github.com/metal-toolbox/firmware-syncer/pull/112/commits/c4ac33fe8a34cd90d53a38467e5f1b30700b84d6)